tests/test_views.py

```diff
@@ -0,0 +1,97 @@
+        factories.SubjectFactory.create(name='sub-1.1', parent=sub_1.id, lft=2, rght=3)
+        response = client.get(reverse('list'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_list.html'
+        assert 'sub-1' in response.context['viewall_list'][0]['hyphenated']
+        assert 'sub-1 - sub-1.1' in response.context['viewall_list'][1]['hyphenated']
+        assert 'sub-2' in response.context['viewall_list'][2]['hyphenated']
+
+
+@pytest.mark.django_db
+class TestBrowseSubjects:
+    def test_browse_subjects(self, client):
+        s1 = factories.SubjectFactory.create()
+        s2 = factories.SubjectFactory.create()
+        response = client.get(reverse('main_browse'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<a href=\'/subjects/browse/{id}/\'>{name}</a>"\
+               .format(id=s1.id, name=s1.name).encode() in response.content
+        assert "<a href=\'/subjects/browse/{id}/\'>{name}</a>"\
+               .format(id=s2.id, name=s2.name).encode() in response.content
+
+    def test_browse_subjects_with_sub_id(self, client):
+        sub = factories.SubjectFactory.create()
+        response = client.get(reverse('browse', kwargs={'sub_id': sub.id}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<input name=\'subject\' type=\'text\' size=\'57\' value=\'{name}\'>".format(
+               name=sub.name).encode() in response.content
+
+    def test_browse_subjects_with_sub_name(self, client):
+        sub = factories.SubjectFactory.create()
+        response = client.get(reverse('browse_sub_name', kwargs={'sub_name': sub.name}))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_browse.html'
+        assert "<input name=\'subject\' type=\'text\' size=\'57\' value=\'{name}\'>".format(
+               name=sub.name).encode() in response.content
+
+
+@pytest.mark.django_db
+class TestSearchSubjects:
+    def test_search_subjects(self, client):
+        subject = factories.SubjectFactory.create()
+        response = client.get(reverse('search_subjects') + '?q=' + subject.name)
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_search.html'
+        assert b'Search for a UNTLBS Subject.' in response.content
+        assert '<span class="subject">{name}</span>'\
+               .format(name=subject.name).encode() in response.content
+
+    def test_empty_search_subjects(self, client):
+        response = client.get(reverse('search_subjects'))
+        assert response.status_code == 200
+        assert response.templates[0].name == 'subjects/subject_search.html'
+        assert b'Please enter a search term.' in response.content
+
+
+@pytest.mark.django_db
+class TestSAboutSubjects:
```

Comment:
Is the "S" after "Test" in this classes name a typo in this and the next test?

Reply:
yes, that was a mistake. I corrected it